### DISCUSSION TITLE
Connection Timeout

### DISCUSSION THREAD

**Discussion by just4give** [timdorr/tesla-api#378] [Q&A]
Hi there,
I was trying to run the example code but getting network timeout. I am not certain if I am passing correct client information. Any thoughts/suggestions? 

<img width="1062" alt="Screen Shot 2021-05-18 at 12 44 57 PM" src="https://user-images.githubusercontent.com/9275193/118692429-0f278800-b7d8-11eb-847c-46b570ac9b4f.png">

**Comment by BlackPhoenix** [comment]
The library currently (if you installed it via Gem) sends a user-agent, which causes Tesla's server to not respond. You'll need to take that user agent out.

I'm a newbie in Ruby, so I'm not sure how to properly prevent the library from sending that header; I understand that you just need to provide it with a parameter but I don't have a sample to provide. Hopefully someone will chime in with a proper example. What I did is overwrite the client.rb file from the library with the latest code here on GitHub.

**Comment by BlackPhoenix** [comment]
Also, where did you get that client_id and client_secret?

**Reply by just4give** [reply]
grabbed them from https://www.teslaapi.io/authentication/oauth  and I got mine working with nodejs ( which I am more familiar with , never used ruby before ) https://github.com/mseminatore/TeslaJS
